src/components/form/PillInput.res

```diff
@@ -0,0 +1,147 @@
+    } else if !CommonAuthUtils.isValidEmail(trimmedItem) {
+      setItems(prev => Array.concat(prev, [trimmedItem]))
+      form.change(name, items->Array.concat([item])->Identity.genericTypeToJson)
+      setInputValue(_ => "")
+      setError(_ => "")
+    } else {
+      setError(_ => "Invalid Email")
+    }
+  }
+
+  let removeItem = itemToRemove => {
+    form.change(name, items->Array.filter(ele => ele !== itemToRemove)->Identity.genericTypeToJson)
+    setItems(prev => prev->Array.filter(item => item != itemToRemove))
+  }
+
+  let saveItem = itemToSave => {
+    let trimmedEditInput = String.trim(editInput)
+    if (
+      duplicateCheck &&
+      trimmedEditInput != itemToSave &&
+      Array.some(items, existingItem => existingItem == trimmedEditInput)
+    ) {
+      setError(_ => "Email already exists")
+      setEditInput(_ => itemToSave)
+    } else if !CommonAuthUtils.isValidEmail(trimmedEditInput) {
+      setItems(prev => prev->Array.map(item => item == itemToSave ? trimmedEditInput : item))
+      let updatedArray =
+        items
+        ->Array.filter(ele => ele !== itemToSave)
+        ->Array.concat([trimmedEditInput])
+      form.change(name, updatedArray->Identity.genericTypeToJson)
+      setEditingItem(_ => None)
+      setEditInput(_ => "")
+      setError(_ => "")
+    } else {
+      setEditInput(_ => itemToSave)
+      setError(_ => "Invalid Email")
+    }
+  }
+
+  let handleKeyDown = e => {
+    open ReactEvent.Keyboard
+    let key = e->key
+    let keyCode = e->keyCode
+    if key === "Enter" || keyCode === 13 {
+      ReactEvent.Keyboard.preventDefault(e)
+      addItem(inputValue)
+    }
+  }
+  // Toggle editing mode for an item
+  let toggleEditingItem = item => {
+    setEditingItem(_ => Some(item))
+    setEditInput(_ => item)
+  }
+
+  <div className="w-full">
+    <div className="w-full flex flex-wrap gap-2 border p-2 text-sm rounded-md">
+      {items
+      ->Array.mapWithIndex((item, i) =>
+        switch editingItem {
+        | Some(selected) if selected == item =>
+          <div key={Int.toString(i)} className="flex flex-wrap gap-1 p-1 border text-sm rounded-md">
+            <input
+              type_="text"
+              value={editInput}
+              onBlur={_ => saveItem(item)}
+              onInput=handleEditChange
+              onKeyDown={ev => {
+                open ReactEvent.Keyboard
+                let key = ev->key
+                let keyCode = ev->keyCode
+                if key === "Enter" || keyCode === 13 {
+                  ReactEvent.Keyboard.preventDefault(ev)
+                  saveItem(item)
+                }
+              }}
+              className="rounded-md p-1 text-sm flex-grow"
+            />
+            <button
+              onClick={_ => removeItem(item)} className="ml-2 hover:bg-gray-300 rounded-md p-1">
+              <Icon name="cross-outline" size=14 />
+            </button>
+          </div>
+        | _ =>
```

Comment:
Add explicitly the None case

Reply:
It's either Some or None. You won't be having more. Correct?

Reply:
Yes Made changes in the switch conditions.